include/bluetooth/audio/csis.h

```diff
@@ -282,20 +282,22 @@ typedef void (*bt_csis_client_lock_changed_cb)(struct bt_csis_client_csis_inst *
 					       bool locked);
 
 /**
- * @typedef bt_csis_client_lock_state_read_cb
- * @brief Callback for bt_csis_client_get_lock_state()
+ * @typedef bt_csis_client_ordered_access_cb_t
+ * @brief Callback for bt_csis_client_ordered_access()
```

Comment:
If the callback is used from `bt_csis_client_ordered_access` context, why it's lifetime is different? I.e. why it's `bt_csis_client_cb` member?

Reply:
Are you asking why this callback is not provided as an argument to `bt_csis_client_ordered_access` instead? 

That is certainly a possibility, but the `bt_csis_client_ordered_access_t` is more likely to change between each call to `bt_csis_client_ordered_access`, whereas `bt_csis_client_ordered_access_cb_t` is more likely the remain the same function. 

e.g. if you want to execute two procedures on an array of set members, you would call `bt_csis_client_ordered_access` with two different callbacks.

Reply:
And then how do you know which procedure is complete?

Reply:
Hmm, `bt_csis_client_ordered_access` could in theory be called with the same array of members and `set_info` at the same time (currently that would fail due to the `busy` flag), and then the callback would be called with the same parameters, yes. 

Currently (due to the `busy` flag) it is not a problem, but I see what you mean. 

Are you suggesting that we switch to use parameter-based callback for all client operations callbacks (like the GATT API)?

Reply:
I guess that way the context would not be lost. From the API user perspective this would be used like this pseudo code:
```
static screen_indicator volume_indicator;

void group_volume_up_complete(group, err)
{
  screen_indicate(screen, screen_indicator, indicate_vol_up);
}

int group_volume_up_func(group/group_member)
{
   return volume_up(group_member);
}

int group_volume_up(group)
{
    return bt_csis_client_ordered_access(group, group_volume_up_func, group_volume_up_complete);
}
```

Reply:
Right. 

I see what you mean, and I see the value of it, especially if we want to enable parallel operations (I don't recall the CSIP specifying that you can't do multiple ordered access procedures in parallel (of course, the "correct" way of doing this would be do do multiple things in the first callback, instead of doing the Ordered Access Procedure multiple times in parallel). 

However, the points you've made here is much more general than just this procedure, and generally applies to all client procedures.

Reply:
I've created an issue here: https://github.com/zephyrproject-rtos/zephyr/issues/44867